packages/x-data-grid/src/hooks/features/rowSelection/useGridRowSelection.ts

```diff
@@ -442,8 +442,13 @@ export const useGridRowSelection = (
   /*
    * EVENTS
    */
+  const isFirstRender = React.useRef(true);
   const removeOutdatedSelection = React.useCallback(
     (sortModelUpdated = false) => {
+      if (isFirstRender.current) {
+        isFirstRender.current = false;
+        return;
+      }
```

Comment:
This could be a useful utility hook if we find ourselves needing to do it in a few places. I have seen a `useUpdateEffect` hook in another codebase that encapsulates this behaviour.

Reply:
`useUpdateEffect` could be helpful yes 👍 
Although in this specific case the function is not executed in an effect, rather it's binded to a Grid event, so `useUpdateEffect` might not work as expected.